core/eolearn/core/eodata_io.py

```diff
@@ -281,20 +303,69 @@ def _save(self, filesystem, data, path, file_format, compress_level=0):
             with gzip.GzipFile(fileobj=file_handle, compresslevel=compress_level, mode='wb') as gzip_file_handle:
                 self._write_to_file(data, gzip_file_handle, file_format)
 
-    @staticmethod
-    def _write_to_file(data, file, file_format):
+    def _write_to_file(self, data, file, file_format):
         """ Writes to a file
         """
         if file_format is FileFormat.NPY:
-            np.save(file, data)
-        elif file_format is FileFormat.PICKLE:
-            pickle.dump(data, file)
+            return np.save(file, data)
+
+        if file_format is FileFormat.GPKG:
+            return data.to_file(file, driver="GPKG", encoding="utf-8", layer=fs.path.basename(self.path))
+
+        if file_format in [FileFormat.JSON, FileFormat.GEOJSON]:
+            if self.feature_type is FeatureType.BBOX:
+                data = data.geojson
 
-    @staticmethod
-    def _decode(file, path):
+            if self.feature_type is FeatureType.TIMESTAMP:
+                data = [timestamp.isoformat() for timestamp in data]
+
+            try:
+                json_data = json.dumps(data, indent=2)
+            except TypeError as exception:
+                raise TypeError(
+                    f'Failed to serialize {self.feature_type} into a JSON file. Make sure that this feature type '
+                    f'contains only basic Python types before attempting to serialize it.'
```

Comment:
I'd write "JSON serializable Python types" instead of "basic Python types".

Reply:
Changed.